src/toil/jobStores/abstractJobStore.py

```diff
@@ -903,6 +904,15 @@ def writeFileStream(self, jobStoreID=None, cleanup=False, basename=None):
                file basename so that when searching the job store with a query
                matching that basename, the file will be detected.
 
+        :param str mode: an optional string that specifies the mode in which the file is opened.
+                It defaults to 'b' which means open for reading in binary mode.
+
+        :param str encoding: the name of the encoding used to decode or encode the file.
+                This should only be used in text mode.
```

Comment:
This should say which encodings are supported; maybe something like "Encodings are the same as for `str.encode()`".

Also, what encoding is the default in text mode?